src/components/TableComponent/index.jsx

```diff
@@ -1,166 +1,120 @@
+    const { sortBy } = args;
+    if (sortBy && sortBy.length > 0) {
+      const column = sortBy[0];
+      const ordering = `${column.desc ? '-' : ''}${column.id}`;
       updateUrl({
         page: 1,
-        ordering: direction === 'desc' ? `-${column.key}` : column.key,
+        ordering,
       });
+      sortTable(ordering);
       sendEnterpriseTrackEvent(enterpriseId, 'edx.ui.enterprise.admin_portal.table.sorted', {
         tableId: id,
-        column: column.label,
-        direction,
+        column: column.id,
+        direction: column.desc ? 'desc' : 'asc',
       });
-    };
-
-    const columnConfig = this.props.columns.map(column => ({
-      ...column,
-      onSort: column.columnSortable ? (direction) => sortByColumn(column, direction) : null,
-    }));
-
-    let sortDirection;
-    let sortColumn;
-
-    if (tableSortable) {
-      sortDirection = ordering && ordering.indexOf('-') !== -1 ? 'desc' : 'asc';
-      sortColumn = (ordering && ordering.replace('-', '')) || columnConfig[0].key;
     }
+  }, []);
 
-    return (
-      <div className={className}>
-        <div className="row">
-          <div className="col">
-            {loading && <TableLoadingOverlay />}
-            <div className="table-responsive">
-              <Table
-                id={id}
-                className="table-sm table-striped"
-                columns={columnConfig}
-                data={formatData(data)}
-                tableSortable={tableSortable}
-                defaultSortedColumn={sortColumn}
-                defaultSortDirection={sortDirection}
-              />
-            </div>
-          </div>
-        </div>
-        <div className="row mt-2">
-          <div className="col d-flex justify-content-center">
-            <Pagination
-              paginationLabel={`${id}-pagination`}
-              pageCount={pageCount}
-              currentPage={currentPage}
-              onPageSelect={(page) => {
-                updateUrl({ page });
-                sendEnterpriseTrackEvent(enterpriseId, 'edx.ui.enterprise.admin_portal.table.paginated', {
-                  tableId: id,
-                  page,
-                });
-              }}
-            />
+  useEffect(() => {
+    // Get initial data
+    paginateTable();
+    return clearTable;
+  }, []);
+
+  const handlePageChange = (page) => {
+    updateUrl({ page });
+    sendEnterpriseTrackEvent(enterpriseId, 'edx.ui.enterprise.admin_portal.table.paginated', {
+      tableId: id,
+      page,
+    });
+    paginateTable(parseInt(page, 10));
+  };
+
+  const renderTableContent = () => (
+    <div className={className}>
+      <div className="row">
+        <div className="col">
+          <div className="table-responsive">
+            <DataTable
+              id={id}
+              className="table-sm table-striped"
+              columns={columns}
+              data={formatData(data)}
+              isSortable={tableSortable}
```

Comment:
Given that some tables support row selection (e.g., code management), I'm not seeing anything regarding the `isSelectable` prop here, which implies the selection logic is custom outside of `DataTable`; ideally, we'd rely on `DataTable`'s selection logic rather than defining our own.